pkg/vpcmodel/explainabilityInput.go

```diff
@@ -107,19 +111,33 @@ func (configsMap MultipleVPCConfigs) getVPCConfigAndSrcDstNodes(src, dst string)
 				errMsgNoValidDst = err
 			}
 		} else {
-			configsWithSrcDstNode[cfgID] = srcAndDstNodes{srcNodes, dstNodes, isSrcDstInternalIP}
+			if configsMap[cfgID].IsMultipleVPCsConfig {
+				configsWithSrcDstNodeMultiVpc[cfgID] = srcAndDstNodes{srcNodes, dstNodes, isSrcDstInternalIP}
+			} else {
+				configsWithSrcDstNodeSingleVpc[cfgID] = srcAndDstNodes{srcNodes, dstNodes, isSrcDstInternalIP}
+			}
 		}
 	}
-	switch len(configsWithSrcDstNode) {
-	case 1: // single match: return it
-		for cfgID, val := range configsWithSrcDstNode {
-			return configsMap[cfgID], val.srcNodes, val.dstNodes, val.isSrcDstInternalIP, nil
-		}
-	case 0:
+	switch {
+	// no match
+	case len(configsWithSrcDstNodeSingleVpc) == 0 && len(configsWithSrcDstNodeMultiVpc) == 0:
 		return noMatchErr(srcFoundSomeCfg, dstFoundSomeCfg, errMsgInternalNoConnectedVSI, errMsgInternalNotWithinSubnet,
 			errMsgNoValidSrc, errMsgNoValidDst)
-	default: // len(configsWithSrcDstNode) > 1: src and dst found in more than one VPC configs - error
-		return configsMap.matchMoreThanOneCfgErr(src, dst, configsWithSrcDstNode)
+	// single match, of a multi vpc config: return it
+	case len(configsWithSrcDstNodeSingleVpc) == 0 && len(configsWithSrcDstNodeMultiVpc) == 1:
+		for cfgID, val := range configsWithSrcDstNodeMultiVpc {
+			return configsMap[cfgID], val.srcNodes, val.dstNodes, val.isSrcDstInternalIP, nil
+		}
+	// Single match of single vpc config, return it. Note that in this case there could be also matches in a
+	// multi vpc config that contains this vpc.
+	case len(configsWithSrcDstNodeSingleVpc) == 1:
+		for cfgID, val := range configsWithSrcDstNodeSingleVpc {
+			return configsMap[cfgID], val.srcNodes, val.dstNodes, val.isSrcDstInternalIP, nil
+		}
+	default: // len(configsWithSrcDstNodeSingleVpc) > 1 ||
```

Comment:
do not understand the logic here and the comments.. (such as "Note that in this case there could be also matches in a
multi vpc config that contains this vpc.")

Reply:
I improved the documentation of this function and of the following one. See if it is clearer now.
I also copied a bug fix from #502 in the following function.